internal/pkg/http/utils.go

```diff
@@ -15,18 +15,34 @@
 package http
 
 import (
+	"bytes"
+	"compress/flate"
+	"compress/gzip"
 	"fmt"
+	"io"
 	"mittens/internal/pkg/placeholders"
 	"strings"
+
+	"github.com/andybalholm/brotli"
```

Comment:
not sure about this last one
are you just trying to cover all bases or is this a requirement that you have today @jrauschenbusch?
if it's not i'd rather keep it out as it seems a bit too niche and requires adding a new dependency.

Reply:
ps: sorry for the delay. holidays got in the way :)

Reply:
Not a requirement, but a standard which becomes more popular as it provides better compression ratio than gzip with nearly same performance. And its focus is on HTTP.

I looked around and this dependency seems to be the defacto standard lib in the Go world.

Reply:
I was sceptical about compression and niche algorithms but this made me think.
We use mittens mostly as a sidecar on a Kubernetes environment. We don't have strict performance requirements. Also, this is communication between containers and does not go through the public internet. However, this may not be the case for others.

How are you using mittens if you can share more details @jrauschenbusch?

Brotli is indeed gaining traction and is now supported end-to-end by [CDN providers](https://blog.cloudflare.com/this-is-brotli-from-origin).

Reply:
We also mainly use mittens as Kubernetes sidecar. But we conducted a lot of tests which led to the result, that we also need to warm up the (de-)compression code (especially dynamic class loading and code JIT compiler actions for JVM-based workloads). We observed a significant higher p90, p95 and p99 latency. We have strict settings for request timeouts. So not warming up leads to a massive amount of timeouts during rolling restarts and scale outs.

Reply:
https://github.com/ExpediaGroup/container-startup-autoscaler is the new startup autoscaler we open sourced today.

Reply:
Yes, this is the issue described in #135.